code/drasil-example/Drasil/GlassBR/ModuleDefs.hs

```diff
@@ -86,41 +88,40 @@ filename = var "filename" "name of file with x y and z data"
 -- Some semantic functions
 
 -- Given two points (x1,y1) and (x2,y2), return the slope of the line going through them
-slope :: (Expr, Expr) -> (Expr, Expr) -> Expr
+slope :: (CodeExpr, CodeExpr) -> (CodeExpr, CodeExpr) -> CodeExpr
```

Comment:
This is very interesting. `slope` is not necessarily a `CodeExpr` generator (though that is indeed how it is used). It kind of wants to be polymorphic over both [though I don't want to go there right now.]

Other routines below really are "pure code".

I don't want you to change anything here, this is just a comment.

Reply:
Thanks, yes, I agree. I believe this particular piece of code originally (pre-me starting the Expr split up) had used the Num typeclass to constrain the points so this code was more generalized then.

Reply:
I'm not yet sure how this piece will look like when we have typed Expr, but hopefully it will revert to the original Num constraint (maybe it was Num, Fractional actually).